src/components/FilterSearch/FilterSearch.tsx

```diff
@@ -37,193 +40,258 @@ interface IFilterSearch {
-          {...params}
-          variant='outlined'
-          className={classes.searchBar}
-          placeholder={selectedFilters.length === 0 ? 'Search token' : ''}
-          InputProps={
-            {
-              ...params.InputProps,
-              style: {
-                padding: 0,
-                height: '100%',
-                display: 'flex',
-                alignItems: 'center',
-                ...typography.body2
-              },
-              endAdornment: (
-                <InputAdornment position='end'>
-                  <img src={SearchIcon} className={classes.searchIcon} alt='Search' />
-                </InputAdornment>
-              ),
-              inputProps: {
-                ...params.inputProps,
-                readOnly: isTokensSelected,
-                style: { paddingLeft: '12px' }
-              },
-              onKeyDown: (event: React.KeyboardEvent<HTMLInputElement>) => {
-                if (params.inputProps.onKeyDown) {
-                  params.inputProps.onKeyDown(event)
-                }
-                if (event.key === 'Backspace' && event.currentTarget.value === '') {
-                  if (isTokensSelected) {
-                  } else if (selectedFilters.length > 0) {
-                    const lastToken = selectedFilters[selectedFilters.length - 1]
-                    handleRemoveToken(lastToken)
-                  }
+    }, [networkType])
+
+    const handleRemoveToken = useCallback(
+      (tokenToRemove: ISearchToken) => {
+        setSelectedFilters(prev => prev.filter(token => token.address !== tokenToRemove.address))
+      },
+      [setSelectedFilters]
+    )
+
+    const handleAutoCompleteChange = useCallback(
+      (_event: any, newValue: ISearchToken[]) => {
+        setSelectedFilters(newValue)
+        setOpen(true)
+      },
+      [setSelectedFilters]
+    )
+
+    const handleOpenPopper = useCallback(() => {
+      if (!isTokensSelected) {
+        setOpen(true)
+      }
+    }, [isTokensSelected])
+
+    const handleClosePopper = useCallback(() => {
+      setOpen(false)
+    }, [])
+
+    const renderOption = useCallback(
+      (autocompleteProps: any, option: ISearchToken) => {
+        const { key, ...rest } = autocompleteProps
+        return (
+          <Box key={key} component='li' {...rest} sx={{ padding: '0 !important' }}>
+            <TokenOption option={option} networkUrl={networkUrl} isSmall={isSmall} />
+          </Box>
+        )
+      },
+      [networkUrl, isSmall]
+    )
+
+    const renderTags = useCallback(
+      (value: ISearchToken[], getTagProps: any) =>
+        value.map((option, index) => (
+          <TokenChip
+            key={option.address}
+            option={option}
+            onRemove={handleRemoveToken}
+            {...getTagProps({ index })}
+          />
+        )),
+      [handleRemoveToken]
+    )
+
+    const handleKeyDown = useCallback(
+      (event: React.KeyboardEvent<HTMLInputElement>, inputProps: any) => {
```

Comment:
please, change types `any` for the correct ones, there are more cases in your code